overwatch/src/services/state.rs

```diff
@@ -153,10 +153,13 @@ pub struct StateHandle<State, Operator> {
     operator: Operator,
 }
 
+// Clone must be used carefully. It's very likely `Operator` will be a
+// `StateOperator`, which are likely behaving in as if they were singletons.
```

Comment:
Why is that? Because they might hold file descriptors? Or what's the case? In any case, if that can happen, do you think it makes sense to introduce another trait to deal with singleton-like operators? Or to let them implement `Clone` in a way that is not "dangerous", so we wouldn't need to specify anything here? It feels we are introducing a lot of cognitive complexities for a developer to be aware of, while maybe we can rely a bit more on the type system?

Reply:
The second alternative is a no-go IMO. Reason is that state is handled by overwatch. But Status is meant for other services to know when a service is ready to be used. I think the `watch` is a better option.